pkg/options/options.go

```diff
@@ -202,5 +204,9 @@ func (o *Options) Validate() error {
 		return fmt.Errorf("value for --auto-gomemlimit-ratio=%f must be greater than 0 and less than or equal to 1", o.AutoGoMemlimitRatio)
 	}
 
+	if o.ObjectLimit < 0 {
+		return fmt.Errorf("value for --object-limit=%d must be equal or greater than 0", o.ObjectLimit)
```

Comment:
Do we want this to be greater than zero? I see in a couple of instances in `watch.go` that we check for this being greater than zero? Or maybe we want the conditions to reflect this value (0) as well, and as such, accommodate this in `watch.go` as well?

The latter would make more sense as in we keep the same expectations as `cache.ListerWatcher` allows upstream.

Reply:
It throws an error because we assume that a negative object limit is invalid. If it's zero, we don't want to throw an error because this is the default and means that it's not set.